utils/led.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * led.c
+ *
+ *  Created on: 29-Jan-2021
+ *      Author: Sonal Shrivastava
+ */
+
+#include <stdint.h>
+//#include <stdio.h>
+//#include <stdlib.h>
+
+#include "drivers/mss_timer/mss_timer.h"
+#include "drivers_config/sys_config/sys_config_mss_clocks.h"
+#include "mss_gpio/mss_gpio.h"
+#include "led.h"
+#include "CMSIS/system_m2sxxx.h"
+
+
+static uint32_t g_gpio_pattern;
+static uint8_t num_of_blinks, count;
+static uint32_t timer2_load_value = MSS_SYS_APB_0_CLK_FREQ;
+
+/* Local function prototypes */
+//static void led_turn_on(void);
+//static void led_turn_off(void);
+static void led_tim2_start(void);
+
+
+
+/**
+ * @brief Initialization function for LED at GPIO0 called upon MSS power on reset
+ *
+ * This function is called is called in main() function at the start to configure
+ * the GPIO0 in OUTPUT_MODE such that this GPIO0 configuration need not be
+ * changed afterwards before calling led_custom_blink().
+ *
+ */
+void led_init(void)
+{
+    /* Initialize MSS GPIOs */
+    MSS_GPIO_init();
+
+    /* Configure MSS GPIO0 in OUTPUT_MODE */
+    MSS_GPIO_config( MSS_GPIO_0 , MSS_GPIO_OUTPUT_MODE);
+
+    g_gpio_pattern = MSS_GPIO_get_outputs();
+
+    g_gpio_pattern &= ~(MSS_GPIO_0_MASK);
+
+    MSS_GPIO_set_outputs(g_gpio_pattern);
+}
+
+
+/**
+ * @brief Custom link function for LED at GPIO0
+ *
+ * This function is called to make LED at GPIO0 (configured as output) blink
+ * a certain number of times defined by the enumeration #led_blink_t present in
+ * led_error.h file.
+ *
+ * @b NOTE:
+ *
+ * - Should be called after led_init() function since it configures
+ * GPIO0 in OUTPUT_MODE upon MSS power on reset.
+ *
+ * @param blinks[in]      Number of blinks for LED
+ *
+ */
+void led_custom_blink(led_blink_t blinks)
+{
+    num_of_blinks = blinks;
+
+    /* Configure TIM2 in periodic mode*/
+    led_tim2_start();
+}
+
+
+
+/**
+ * @brief Configure TIM2 in periodic mode
+ *
+ * @b NOTE:
+ *
+ * - This function passes a pre-defined delay value to be loaded into the timer
+ * which is equal to 500ms.
+ *
+ */
+void led_tim2_start(void)
```

Comment:
The function declaration should also be qualified as `static`, same as the prototype.

Reply:
We are not going to need this function anyway. Because `delay()` fn worked. And i have turned `TIM1` into a 64-bit TIM. So, we do not need function for LED timer.